### PR TITLE
Gzip: add note about file compression

### PR DESCRIPTION
**Pull request by thumphries**
As far as I can tell, the `Data.Default` instance for `GzipSettings` does nothing. More than half of the Github results appear to be [using this accidentally](https://github.com/search?q=language%3Ahaskell+%22gzip+def%22&ref=simplesearch&type=Code&utf8=%E2%9C%93), and getting no compression as a result. I had to look at the source after testing my application to realise this instance was not useful.

Could also fix via improved documentation (or providing some `gzipCompress :: GzipSettings` that is easy to discover). However, I am struggling to think of cases where you'd import `Network.Wai.Middleware.Gzip` and actually want a no-op middleware, so I am suggesting a change to the instance.

### CODE REVIEW

**Comment by thumphries** [comment]
Apologies if I've misunderstood something :)

**Comment by snoyberg** [comment]
The default middleware right now is not a no op. It simply doesn't compress files, which is a sensible default since file compression disabled the send file optimization

**Comment by thumphries** [comment]
Ah, that makes sense. I was only checking the compression of static JS resources.

I'd like to add a comment to that effect, will update this PR.

**Comment by thumphries** [comment]
Updated docs. Let me know if anything is wrong.

**Comment by erikd** [comment]
:+1: LGTM!

File: wai-extra/Network/Wai/Middleware/Gzip.hs [hunk 1]
```diff
@@ -50,12 +50,23 @@ data GzipSettings = GzipSettings
...
+                   -- platforms.
+    | GzipCacheFolder FilePath -- ^ Compress files, caching them in
+                               -- some directory.
     deriving (Show, Eq, Read)
 
+-- | Use default MIME settings; /do not/ compress files.
 instance Default GzipSettings where
     def = GzipSettings GzipIgnore defaultCheckMime
 
+-- | MIME types that will be compressed by default:
```
Comment (snoyberg) [inline wai-extra/Network/Wai/Middleware/Gzip.hs]: This code will also compress anything that starts with `text/`.

**Comment by snoyberg** [comment]
Looks really good, I only had one minor comment.

**Comment by thumphries** [comment]
> This code will also compress anything that starts with text/.

I truly am blind. Fixed

**Comment by snoyberg** [comment]
Awesome, thank you!

**Pull request merged**